components/react/src/App.js

```diff
@@ -1,56 +1,58 @@
-import React, { Component } from 'react';
+import React, { useEffect, useState } from 'react';
 import Header from './Components/Header';
 import ProjectList from './Components/ProjectList';
-import SelectorMediator from './Components/SelectorMediator';
-
-class App extends Component {
-  abortController = new AbortController();
-
-  constructor(props) {
-    super(props);
-    this.state = {
-      isLoaded: false,
-      projects: [],
-    };
-  }
-
-  componentDidMount() {
-    this.request('http://localhost:5455/projects.json', 'projects')
-      .then(() => this.request('http://localhost:5455/taxonomies.json', 'taxonomies'))
-      .then(() => this.setState(Object.assign({}, this.state, { isLoaded: true })));
-  }
-
-  request(url, stateVariable) {
-    return fetch(url, { signal: this.abortController.signal })
+import SelectorList from './Components/SelectorList';
+
+const App = () => {
+  const abortController = new AbortController();
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [projects, setProjects] = useState([]);
+  const [taxonomies, setTaxonomies] = useState([]);
+  const [error, setError] = useState();
+  const [selected, setSelected] = useState({});
+  const [matchScores, setMatchScores] = useState({});
+
+  useEffect(
+    () => request('projects', setProjects).then(() => request('taxonomies', setTaxonomies)).then(() => setIsLoaded(true)),
+    [],
+  );
+  const request = (endpoint, setter) => {
+    return fetch(`http://localhost:5455/${endpoint}.json`, { signal: abortController.signal })
       .then(result => result.json())
-      .then(result => this.addResultToState(result, stateVariable), this.handleFetchError);
-  }
+      .then(setter, handleFetchError);
+  };
 
-  addResultToState(result, stateVariable) {
-    this.setState(Object.assign({}, this.state, { [stateVariable]: result }));
-  }
-
-  handleFetchError(error) {
+  const handleFetchError = error => {
     if (error.name === 'AbortError') return;
-    this.setState({ isLoaded: true, error });
+    setIsLoaded(true);
+    setError(error);
   }
 
-  render() {
-    const { error, isLoaded } = this.state;
-    if (error) {
-      return <div className="App"><Header />Error: {error.message}</div>;
-    } else if (!isLoaded) {
-      return <div className="App"><Header />Loading...</div>;
-    } else {
-      return (
-        <div className="App">
-          <Header />
-          <SelectorMediator {...this.state} />
-          <ProjectList {...this.state} />
+  const generateMatch = () => fetch(
+    'http://localhost:5455/matches.json',
```

Comment:
Same comment, localhost:5455 only works locally.

Reply:
Fixed, now it is an environment variable called `API_URL`.